media/rtm_change_upload_directory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * This snippet allows you to change upload directory to /wp-content/uploads/rtMedia/users/userid/postid/filename.
+ * The media must be uploaded through rtMedia uploader (calling `rtmedia_filter_upload_dir` filter).
+ * This code will only work if the media is being uploaded from the post, i.e. if `get_the_ID()` returns the valid post ID, it won't work if you upload from the profile.
+ *
+ * All your previously uploaded media files won't be affected by this code, but new media files will be uploaded to specified directory.
+ */
+
+// Check whether function exists or not.
+if ( ! function_exists( 'rtmedia_set_upload_path_to_post_id' ) ) {
+
+	/**
+	 * This filter is called when a media is being uploaded through rtMedia upload mechanism.
+	 * This filter is used to change the upload directory.
+	 */
+	add_filter( 'rtmedia_filter_upload_dir', 'rtmedia_set_upload_path_to_post_id', 10, 1 );
```

Comment:
FYI -- `1` is the default value `$accepted_args` argument. So you don't need to pass default value.